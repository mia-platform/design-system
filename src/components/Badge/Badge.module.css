.badge {
  display: flex;
  gap: var(--spacing-gap-lg, 16px);
  align-items: center;
  width: 100%;
  max-width: inherit;
  padding: var(--spacing-padding-lg, 16px);
  background: var(--Grey-200, #F2F2F2);
  border-radius: var(--shape-border-radius-lg, 8px);

  & .icon {
    width: var(--shape-size-2xl, 48px);
    height: var(--shape-size-2xl, 48px);
    flex-shrink: 0;
  }

  & .titleWrapper {
    display: flex;
    flex-direction: column;
    gap: var(--spacing-gap-xs, 4px);
    flex-shrink: 1;
    overflow: hidden;

    & .title {
      display: flex;
      align-items: center;
      gap: var(--spacing-gap-sm, 8px);
    }
  }

  & .extra {
    display: flex;
    align-items: center;
    justify-content: flex-end;
    flex-grow: 1;
  }
}

.tooltip {
  min-height: 100px;
  max-height: 300px;
  overflow: auto;
}
