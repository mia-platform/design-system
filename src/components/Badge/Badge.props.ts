/**
 * Copyright 2024 Mia srl
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import { ReactNode } from 'react'

import { IconComponent } from '../Icon/Icon.props'

export type BadgeProps = {

  /**
   * Optional description, it displays formatted text customized content below the badge title.
   */
  description?: string | ReactNode;

  /**
   * Additional content to be displayed on the right side of the badge.
   */
  extra?: ReactNode;

  /**
   * The icon component to render within the badge.
   */
  icon: IconComponent;

  /**
   * The main title text, it displays formatted text or customized content within the badge.
   */
  title: string | ReactNode;

  /**
   * Whether the badge description should be multiline or clipped with ellipsis at the first line.
   * */
  hasMultilineDescription?: boolean;

  /**
   * Additional content to be displayed alongside the badge title (e.g., a tag or an icon).
   */
  titleExtra?: ReactNode;
};
