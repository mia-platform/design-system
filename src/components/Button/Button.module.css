.button {
  display: flex;
  align-items: center;
  justify-content: center;
  gap: calc(var(--spacing-gap-xs, 4) * 1px);
}

/* TODO: add 12px padding to theme */
.buttonSm {
  padding-top: calc(var(--spacing-padding-xs, 4) * 1px - 1px) !important;
  padding-right: calc((var(--spacing-padding-xs, 4) + var(--spacing-padding-sm, 8)) * 1px - 1px) !important;
  padding-bottom: calc(var(--spacing-padding-xs, 4) * 1px - 1px) !important;
  padding-left: calc((var(--spacing-padding-xs, 4) + var(--spacing-padding-sm, 8)) * 1px - 1px) !important;
}

.buttonSmIconOnly {
  padding: calc(var(--spacing-padding-xs, 4) * 1px - 1px) !important;
}

.buttonMd {
  padding-top: calc(var(--spacing-padding-sm, 8) * 1px - 1px) !important;
  padding-right: calc((var(--spacing-padding-xs, 4) + var(--spacing-padding-sm, 8)) * 1px - 1px) !important;
  padding-bottom: calc(var(--spacing-padding-sm, 8) * 1px - 1px) !important;
  padding-left: calc((var(--spacing-padding-xs, 4) + var(--spacing-padding-sm, 8)) * 1px - 1px) !important;
}

.buttonMdIconOnly {
  padding: calc(var(--spacing-padding-sm, 8) * 1px - 1px) !important;
}

.buttonLg {
  height: calc(var(--shape-size-4xl, 48) * 1px) !important;
  padding: calc(var(--spacing-padding-md, 16) * 1px - 1px) !important;
}

.buttonGhost {
  border-color: transparent !important;
  background-color: transparent;
}

.buttonText {
  padding: 0 calc(var(--spacing-padding-xs, 4) * 1px);
}
