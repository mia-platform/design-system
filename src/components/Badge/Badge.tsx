/**
 * Copyright 2024 Mia srl
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import { ReactElement, useMemo } from 'react'

import { BadgeProps } from './Badge.props'
import { Icon } from '../Icon'
import { Typography } from '../Typography'
import styles from './Badge.module.css'
import { useTheme } from '../../hooks/useTheme'

/**
 * UI component for displaying information about a resource, emphasizing some of its main features.
 * @returns {ReactElement} Badge component
 */
export const Badge = ({
  description: customDescription,
  extra: customExtra,
  icon: customIcon,
  title: customTitle,
  titleExtra,
}: BadgeProps): ReactElement => {
  const { palette } = useTheme()

  const icon = useMemo(() => (
    <div className={styles.icon} data-testid="badge-icon">
      <Icon color={palette.text.neutral.subtle} component={customIcon} size={48} />
    </div>
  ), [customIcon, palette.text.neutral.subtle])

  const title = useMemo(() => (
    <div className={styles.title}>
      <Typography.H3 ellipsis={{ rows: 1, tooltip: { title: customTitle } }}>
        {customTitle}
      </Typography.H3>
      {titleExtra}
    </div>
  ), [customTitle, titleExtra])

  const description = useMemo(() => {
    if (!customDescription) { return }

    return (
      <Typography.BodyS ellipsis={{ rows: 1, tooltip: { title: customDescription } }}>
        {customDescription}
      </Typography.BodyS>
    )
  }, [customDescription])

  const extra = useMemo(() => {
    if (!customExtra) { return }

    return (
      <div className={styles.extra}>
        {customExtra}
      </div>
    )
  }, [customExtra])

  return (
    <div className={styles.badge}>
      {icon}
      <div className={styles.content}>
        {title}
        {description}
      </div>
      {extra}
    </div>
  )
}
