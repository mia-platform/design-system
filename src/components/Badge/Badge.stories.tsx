/**
 * Copyright 2024 Mia srl
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import type { Meta, StoryObj } from '@storybook/react'
import { PiCircleHalfTilt } from 'react-icons/pi'

import { Badge } from './Badge'
import { Button } from '../Button'
import { Tag } from '../Tag'
import { loremIpsum } from '../Typography/Typography.mocks'

const meta = {
  component: Badge,
} satisfies Meta<typeof Badge>

export default meta
type Story = StoryObj<typeof meta>

const defaults = {
  icon: PiCircleHalfTilt,
  description: loremIpsum,
  title: 'Title',
}

export const Default: Story = {
  args: {
    ...defaults,
  },
  decorators: [
    (_, { args }) => {
      return (
        <Badge {...args} />
      )
    },
  ],
}

export const TitleExtra: Story = {
  args: {
    ...defaults,
    titleExtra: <Tag>Tag</Tag>,
  },
  decorators: [
    (_, { args }) => {
      return (
        <Badge {...args} />
      )
    },
  ],
}

export const CustomTitle: Story = {
  args: {
    ...defaults,
    title: <span style={{ color: 'red', fontWeight: '600' }}>Custom title</span>,
  },
  decorators: [
    (_, { args }) => {
      return (
        <Badge {...args} />
      )
    },
  ],
}

export const CustomSubtitle: Story = {
  args: {
    ...defaults,
    description: <span style={{ color: 'blue', textDecoration: 'underline' }}>Custom description</span>,
  },
  decorators: [
    (_, { args }) => {
      return (
        <Badge {...args} />
      )
    },
  ],
}

export const Extra: Story = {
  args: {
    ...defaults,
    extra: <Button>Button</Button>,
  },
  decorators: [
    (_, { args }) => {
      return (
        <Badge {...args} />
      )
    },
  ],
}

