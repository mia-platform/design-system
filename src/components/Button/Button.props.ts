/**
 * Copyright 2023 Mia srl
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import { MouseEvent, ReactNode } from 'react'

import { Hierarchy, IconPosition, Shape, Size, Type } from './Button.types'

export type ButtonProps = {

  /**
   * The children nodes to be rendered within the button context.
   */
  children?: ReactNode,

  /**
   * Defines the button hierarchy. Either:
   *
   * - "primary": button associated with the most significant (and therefore primary) action on the page;
   * - "neutral": button associated with a general purpose action;
   * - "danger": button associated with a potentially dangerous action.
   */
  hierarchy?: Hierarchy,

  /**
   * Redirect url of a link button.
   * For security reasons, the attribute "rel: 'noopener noreferrer'" is always specified.
   */
  href?: string,

  /**
   * Sets an icon for the button component.
   */
  icon?: ReactNode,

  /**
   * Defines a position for the button icon (if any). Either:
   *
   * - "left": the icon is placed to the left of the button text;
   * - "right": the icon is placed to the right of the button text.
   */
  iconPosition?: IconPosition,

  /**
   * Identifies whether the button is disabled or not.
   */
  isDisabled?: boolean,

  /**
   * Sets the loading status of the button.
   */
  isLoading?: boolean,

  /**
   * Sets the handler to handle a click event.
   */
  onClick?: (event: MouseEvent) => void,

  /**
   * Defines the button shape. Either:
   *
   * - "square": square-shaped button with rounded corners;
   * - "circle": fully rounded button, typically used for buttons containing only icon.
   */
  shape?: Shape,

  /**
   * Defines the button size. Either:
   *
   * - "small": small button - 24px in height by default;
   * - "middle": middle button - 32px in height by default;
   * - "large": large button - 48px in height by default.
   */
  size?: Size,

  /**
   * Specifies where the linked document will open when the link is clicked.
   * Only usable for link buttons together with the href property.
   *
   * @link https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a
   */
  target?: string,

  /**
   * Defines the button type. Either:
   *
   * - "filled": button with a solid fill, typically used for primary actions;
   * - "outlined": medium-emphasis button with a stroke around the button container and no fill,
   *   typically used for secondary actions;
   * - "ghost": transparent button with no stroke, typically used for links.
   */
  type?: Type,
}
