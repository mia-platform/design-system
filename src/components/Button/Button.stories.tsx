/**
 * Copyright 2023 Mia srl
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import type { Meta, StoryObj } from '@storybook/react'
import { action } from '@storybook/addon-actions'

import { ButtonHierarchies, ButtonIconPositions, ButtonShapes, ButtonSizes, ButtonTypes } from './Button.types'
import { Button } from '.'
import { Icon } from '../Icon'

const { Primary, Neutral, Danger } = ButtonHierarchies
const { Left, Right } = ButtonIconPositions
const { Square, Circle } = ButtonShapes
const { Small, Middle, Large } = ButtonSizes
const { Filled, Outlined, Ghost } = ButtonTypes

const icon = <Icon color="white" name="PiCircleHalfTiltLight" size={16} />

const meta = {
  component: Button,
  args: {
    ...Button.defaultProps,
    children: 'Button',
    onClick: action('click'),
  },
} satisfies Meta<typeof Button>

export default meta
type Story = StoryObj<typeof meta>

export const PrimaryFilled: Story = {
  args: {
    ...meta.args,
    hierarchy: Primary,
    type: Filled,
  },
}

export const PrimaryOutline: Story = {
  args: {
    ...meta.args,
    hierarchy: Primary,
    type: Outlined,
  },
}

export const PrimaryGhost: Story = {
  args: {
    ...meta.args,
    hierarchy: Primary,
    type: Ghost,
  },
}

export const NeutralOutline: Story = {
  args: {
    ...meta.args,
    hierarchy: Neutral,
    type: Outlined,
  },
}

export const NeutralGhost: Story = {
  args: {
    ...meta.args,
    hierarchy: Neutral,
    type: Ghost,
  },
}

export const DangerFilled: Story = {
  args: {
    ...meta.args,
    hierarchy: Danger,
    type: Filled,
  },
}

export const DangerOutline: Story = {
  args: {
    ...meta.args,
    hierarchy: Danger,
    type: Outlined,
  },
}

export const DangerGhost: Story = {
  args: {
    ...meta.args,
    hierarchy: Danger,
    type: Ghost,
  },
}

export const SquareShape: Story = {
  args: {
    ...meta.args,
    children: undefined,
    icon,
    shape: Square,
  },
}

export const CircleShape: Story = {
  args: {
    ...meta.args,
    children: undefined,
    icon,
    shape: Circle,
  },
}

export const SmallSize: Story = {
  args: {
    ...meta.args,
    size: Small,
  },
}

export const MiddleSize: Story = {
  args: {
    ...meta.args,
    size: Middle,
  },
}

export const LargeSize: Story = {
  args: {
    ...meta.args,
    size: Large,
  },
}

export const Disabled: Story = {
  args: {
    ...meta.args,
    isDisabled: true,
  },
}

export const Loading: Story = {
  args: {
    ...meta.args,
    isLoading: true,
  },
}

export const WithIconLeft: Story = {
  args: {
    ...meta.args,
    icon,
    iconPosition: Left,
  },
}

export const WithIconRight: Story = {
  args: {
    ...meta.args,
    icon,
    iconPosition: Right,
  },
}
