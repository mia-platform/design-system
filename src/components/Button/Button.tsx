/**
 * Copyright 2023 Mia srl
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import { ReactElement, ReactNode, useMemo } from 'react'
import { Button as AntButton } from 'antd'
import classnames from 'classnames'

import { ButtonHierarchies, ButtonIconPositions, ButtonShapes, ButtonSizes, ButtonTypes } from './Button.types'
import styles from './Button.module.css'

const { button, buttonSm, buttonSmIconOnly, buttonMd, buttonMdIconOnly, buttonLg, buttonGhost, buttonText } = styles

const { Primary, Neutral, Danger } = ButtonHierarchies
const { Left, Right } = ButtonIconPositions
const { Square } = ButtonShapes
const { Small, Middle, Large } = ButtonSizes
const { Filled, Ghost } = ButtonTypes

export type ButtonProps = {

  /**
   * The children nodes to be rendered within the button context.
   */
  children?: ReactNode,

  /**
   * Defines the button hierarchy. Either:
   *
   * - primary: button associated with the most significant (and therefore primary) action on the page;
   * - neutral: button associated with a general purpose action;
   * - danger: button associated with a potentially dangerous action.
   */
  hierarchy?: ButtonHierarchies,

  /**
   * Redirect url of a link button.
   * For security reasons, the attribute "rel: 'noopener noreferrer'" is always specified.
   */
  href?: string,

  /**
   * Sets an icon for the button component.
   */
  icon?: ReactNode,

  /**
   * Defines a position for the button icon (if any). Either:
   *
   * - left: the icon is placed to the left of the button text;
   * - right: the icon is placed to the right of the button text.
   */
  iconPosition?: ButtonIconPositions,

  /**
   * Identifies whether the button is disabled or not.
   */
  isDisabled?: boolean,

  /**
   * Sets the loading status of the button.
   */
  isLoading?: boolean,

  /**
   * Sets the handler to handle a click event.
   */
  onClick?: () => void,

  /**
   * Defines the button shape. Either:
   *
   * - square: square-shaped button with rounded corners;
   * - circle: fully rounded button, typically used for buttons containing only icon.
   */
  shape?: ButtonShapes,

  /**
   * Defines the button size. Either:
   *
   * - small: small button - 24px in height by default;
   * - middle: middle button - 32px in height by default;
   * - large: large button - 48px in height by default.
   */
  size?: ButtonSizes,

  /**
   * Specifies where the linked document will open when the link is clicked.
   * Only usable for link buttons together with the href property.
   *
   * @link https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a
   */
  target?: string,

  /**
   * Defines the button type. Either:
   *
   * - filled: button with a solid fill, typically used for primary actions;
   * - outlined: medium-emphasis button with a stroke around the button container and no fill,
   *   typically used for secondary actions;
   * - ghost: transparent button with no stroke, typically used for links.
   */
  type?: ButtonTypes,
}

/**
 * UI component for performing actions on the page interacting through clicks
 *
 * @link https://ant.design/components/button
 * @returns {Button} Button component
 */
export const Button = ({
  children,
  hierarchy,
  href,
  icon,
  iconPosition,
  isDisabled,
  isLoading,
  onClick,
  shape,
  size,
  target,
  type,
}: ButtonProps): ReactElement => {
  const buttonClassNames = useMemo(() => classnames(
    [
      button,
      children && size === Small && buttonSm,
      !children && size === Small && buttonSmIconOnly,
      children && size === Middle && buttonMd,
      !children && size === Middle && buttonMdIconOnly,
      size === Large && buttonLg,
      type === Ghost && buttonGhost,
    ]
  ), [children, size, type])

  return (
    <AntButton
      className={buttonClassNames}
      danger={hierarchy === Danger}
      disabled={isDisabled}
      ghost={type !== Filled && hierarchy !== Neutral}
      loading={isLoading}
      shape={shape === Square ? 'default' : 'circle'}
      size={size}
      type={hierarchy === Neutral ? 'default' : 'primary'}
      onClick={onClick}
      {...href && { href, rel: 'noopener noreferrer', target }}
    >
      {iconPosition === Left && icon}
      {children && <div className={buttonText}>{children}</div>}
      {iconPosition === Right && icon}
    </AntButton>
  )
}

Button.defaultProps = {
  hierarchy: Primary,
  iconPosition: Left,
  isDisabled: false,
  isLoading: false,
  shape: Square,
  size: Middle,
  type: Filled,
}
