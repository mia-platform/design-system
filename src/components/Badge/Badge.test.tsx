/**
 * Copyright 2024 Mia srl
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import { PiCircleHalfTilt } from 'react-icons/pi'

import { render, screen } from '../../test-utils'
import { Badge } from './Badge'
import { Button } from '../Button'

describe('Badge Component', () => {
  beforeEach(() => {
    jest.resetAllMocks()
  })

  test('renders a badge with icon and title', () => {
    const { asFragment } = render(<Badge icon={PiCircleHalfTilt} title="Title" />)

    expect(screen.getByText('Title')).toBeVisible()
    expect(asFragment()).toMatchSnapshot()
  })

  test('renders a badge with description', () => {
    const { asFragment } = render(
      <Badge
        description="Description"
        icon={PiCircleHalfTilt}
        title="Title"
      />
    )

    expect(screen.getByRole('paragraph')).toBeInTheDocument()
    expect(asFragment()).toMatchSnapshot()
  })

  test('renders a badge with an extra', () => {
    const { asFragment } = render(
      <Badge
        description="Description"
        extra={<Button>Button</Button>}
        icon={PiCircleHalfTilt}
        title="Title"
      />
    )

    expect(screen.getByText('Button')).toBeInTheDocument()
    expect(asFragment()).toMatchSnapshot()
  })
})
