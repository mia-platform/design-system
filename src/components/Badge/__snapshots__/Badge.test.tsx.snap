// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge Component renders a badge with an extra 1`] = `
<DocumentFragment>
  <div
    class="badge"
  >
    <div
      class="icon"
      data-testid="badge-icon"
    >
      <svg
        color="#898989"
        fill="currentColor"
        height="48"
        role="img"
        stroke="currentColor"
        stroke-width="0"
        style="color: rgb(137, 137, 137);"
        viewBox="0 0 256 256"
        width="48"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M201.54,54.46A104,104,0,0,0,54.46,201.54,104,104,0,0,0,201.54,54.46ZM184,195.87a87.16,87.16,0,0,1-16,10.5V99.31l16-16Zm-80-32.56,16-16v68.28a88.37,88.37,0,0,1-16-3ZM88,206.37a87,87,0,0,1-16.3-10.76L88,179.31Zm48-75.06,16-16v97.32a88.37,88.37,0,0,1-16,3ZM40,128A88,88,0,0,1,184.3,60.39L60.38,184.31A87.34,87.34,0,0,1,40,128Zm160,50.59V77.41a88,88,0,0,1,0,101.18Z"
        />
      </svg>
    </div>
    <div
      class="content"
    >
      <div
        class="title"
      >
        <h3
          aria-label="Title"
          class="mia-platform-typography mia-platform-typography-ellipsis mia-platform-typography-ellipsis-single-line h3"
          role="h3"
        >
          Title
        </h3>
      </div>
      <div
        class="mia-platform-typography mia-platform-typography-ellipsis mia-platform-typography-ellipsis-single-line bodyS"
        role="paragraph"
      >
        Description
      </div>
    </div>
    <div
      class="extra"
    >
      <button
        class="mia-platform-btn mia-platform-btn-primary mia-platform-btn-color-primary mia-platform-btn-variant-solid button buttonMd"
        type="button"
      >
        <div
          class="buttonText"
        >
          Button
        </div>
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Badge Component renders a badge with description 1`] = `
<DocumentFragment>
  <div
    class="badge"
  >
    <div
      class="icon"
      data-testid="badge-icon"
    >
      <svg
        color="#898989"
        fill="currentColor"
        height="48"
        role="img"
        stroke="currentColor"
        stroke-width="0"
        style="color: rgb(137, 137, 137);"
        viewBox="0 0 256 256"
        width="48"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M201.54,54.46A104,104,0,0,0,54.46,201.54,104,104,0,0,0,201.54,54.46ZM184,195.87a87.16,87.16,0,0,1-16,10.5V99.31l16-16Zm-80-32.56,16-16v68.28a88.37,88.37,0,0,1-16-3ZM88,206.37a87,87,0,0,1-16.3-10.76L88,179.31Zm48-75.06,16-16v97.32a88.37,88.37,0,0,1-16,3ZM40,128A88,88,0,0,1,184.3,60.39L60.38,184.31A87.34,87.34,0,0,1,40,128Zm160,50.59V77.41a88,88,0,0,1,0,101.18Z"
        />
      </svg>
    </div>
    <div
      class="content"
    >
      <div
        class="title"
      >
        <h3
          aria-label="Title"
          class="mia-platform-typography mia-platform-typography-ellipsis mia-platform-typography-ellipsis-single-line h3"
          role="h3"
        >
          Title
        </h3>
      </div>
      <div
        class="mia-platform-typography mia-platform-typography-ellipsis mia-platform-typography-ellipsis-single-line bodyS"
        role="paragraph"
      >
        Description
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Badge Component renders a badge with icon and title 1`] = `
<DocumentFragment>
  <div
    class="badge"
  >
    <div
      class="icon"
      data-testid="badge-icon"
    >
      <svg
        color="#898989"
        fill="currentColor"
        height="48"
        role="img"
        stroke="currentColor"
        stroke-width="0"
        style="color: rgb(137, 137, 137);"
        viewBox="0 0 256 256"
        width="48"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M201.54,54.46A104,104,0,0,0,54.46,201.54,104,104,0,0,0,201.54,54.46ZM184,195.87a87.16,87.16,0,0,1-16,10.5V99.31l16-16Zm-80-32.56,16-16v68.28a88.37,88.37,0,0,1-16-3ZM88,206.37a87,87,0,0,1-16.3-10.76L88,179.31Zm48-75.06,16-16v97.32a88.37,88.37,0,0,1-16,3ZM40,128A88,88,0,0,1,184.3,60.39L60.38,184.31A87.34,87.34,0,0,1,40,128Zm160,50.59V77.41a88,88,0,0,1,0,101.18Z"
        />
      </svg>
    </div>
    <div
      class="content"
    >
      <div
        class="title"
      >
        <h3
          aria-label="Title"
          class="mia-platform-typography mia-platform-typography-ellipsis mia-platform-typography-ellipsis-single-line h3"
          role="h3"
        >
          Title
        </h3>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
